runtime/sema/check_composite_declaration.go

```diff
@@ -819,6 +843,15 @@ func (checker *Checker) declareCompositeLikeMembersAndValue(
 			nestedCompositeDeclarationVariable :=
 				checker.valueActivations.Find(identifier.Identifier)
 
+			if identifier.Identifier == ast.ResourceDestructionDefaultEventName {
+				// Find the default event's type declaration
+				defaultEventType :=
+					checker.typeActivations.Find(identifier.Identifier)
+				defaultEventComposite := defaultEventType.Type.(*CompositeType)
+				compositeType.DefaultDestroyEvent = defaultEventComposite
+				return
```

Comment:
Should this really return from the whole function?

Reply:
No, removed